handlers/messageUpdate.ts

```diff
@@ -0,0 +1,21 @@
+import { Client, Message } from 'discord.js';
+
+/**
+ * @handler Message Events
+ * @related messageUpdate
+ */
+export async function handleMessageUpdateEvent(client: Client, oldMessage: Message, newMessage: Message) {
+    /**
+     * @event messagePinned
+     * @description Emitted when a message has been pinned.
+     * @param {DJS:Message} message The message who has been pinned.
+     * @param {boolean} oldMessage The old message.
+     * @param {boolean} newMessage The new message.
+     * @example
+     * client.on("messagePinned", (message) => {
+     *   console.log("This message bas been pinned : "+message);
+     * });
+     */
+    if (oldMessage.pinned !== true && newMessage.pinned === true) {
+        client.emit('messagePinned',newMessage.content);
```

Comment:
Il vaut mieux passer le message en tant qu'objet:
```suggestion
        client.emit('messagePinned', newMessage);
```